scanpy/_utils/__init__.py

```diff
@@ -43,6 +44,13 @@
 from .._settings import settings
 from .compute.is_constant import is_constant  # noqa: F401
 
+if Version(anndata_version) >= Version("0.10.0"):
+    from anndata._core.sparse_dataset import (
+        BaseCompressedSparseDataset as SparseDataset,
+    )
```

Comment:
Codecov says this isn’t covered. Is it wrong? We should definitely be testing with the newest anndata version.

Reply:
Well I don't know if it's wrong, but locally when I have the right version on `AnnData` installed, it works.  Has `codecov` ever complained about `TYPE_CHECKING` blocks before?

Reply:
It’s supposed to not care about them: https://github.com/scverse/scanpy/blob/8fe4c3a32bb931d1b5760e3982f03cae805bd88c/pyproject.toml#L214

Reply:
<img width="1512" alt="Screenshot 2024-05-17 at 15 03 33" src="https://github.com/scverse/scanpy/assets/43999641/7ed6d5c0-001d-4f38-9617-4abcb41fe25c">

for the latest `anndata` and

<img width="1511" alt="Screenshot 2024-05-17 at 15 07 16" src="https://github.com/scverse/scanpy/assets/43999641/8379881b-8ce9-4ee0-b3c8-8b5ad187a647">

for `anndata < 0.10.0`, so there seems to be something up with the CI unless I'm missing something.  It's especially strange that this is now the only block missing it (i.e., even the below block is not missing). 

Also, very unhelpful that I can't look on codecov: https://app.codecov.io/gh/scverse/scanpy/commit/2e8ba990c7f030584d9a3a5d10330f1de49759c4/blob/scanpy/_utils/__init__.py

Reply:
UPDATE: nevermind, I found a different way to view it and codecov claims this is actually covered https://app.codecov.io/gh/scverse/scanpy/pull/3048/blob/scanpy/_utils/__init__.py

Reply:
OK, great, that should be all